torchdynamo/symbolic_convert.py

```diff
@@ -545,6 +546,10 @@ def FOR_ITER(self, inst):
 
     def COMPARE_OP(self, inst):
         left, right = self.popn(2)
+        if isinstance(left, UnspecializedPythonVariable):
+            left = left.convert_to_constant(self)
```

Comment:
For everyplace calling ```convert_to_constant```, potentially there is chance this variable has been used before this, which makes the same variable has multiple different copies. But this never occurs for all torchbench models, so I leave it as is. The future plan is: I have another PR which use guard failures to determinate switch between spec/unspec, there is a global data structure to store all local variables and distinct values occurred. When it's ready to post, I'll update this function to first check if this variable exists in the global data structure and:  1/ convert to constant if not; 2/ raise ```RestartAnalysis``` and mark this as _must spec_ if yes.